x-pack/plugins/security_solution/server/lib/entity_analytics/entity_store/definition.ts

```diff
@@ -0,0 +1,142 @@
+Note that it is likely that one entity definition will need to be created for each entity type, namely host and user entities.
+
+
+[x] Utilizes host.name and user.name for the unique identifier
+
+[x] Captures the metadata fields determined as a result of this spike, 
+[ ] ensuring that array fields are captured as arrays, and singular fields capture only the latest value
+
+[ ] Incorporates the index patterns found in the security default data view as a set of data sources
+
+[ ] Incorporates the latest risk score and latest asset criticality as data sources for entity attribute maintenance
+A note on asset criticality: due to the fact that this internal index is not currently ECS compliant, we will have to include a filter that ensures the id_field is only for the appropriate entity type, such as host or user. This issue will alleviate that need in the future.
+
+[ ] Utilizes the @timestamp field // For what?
+
+[x] Defines a history and latest transform that run once per minute
+
+
+
+'{
+  "id": "TODO___host_entity-store",
+  "name": "EA Host Store",
+  "type": "node",
+  "indexPatterns": [...DEFAULT_INDEX_PATTERN],
+  "filter": "@timestamp >= now-5m",
+  "lookback": "5m",
+  "identityFields": [
+    "host.name",
+  ],
+  "displayNameTemplate": "{{host.name}}",
+  "metadata": [
+    "host.architecture",
+    "host.domain",
+    "host.hostname",
+    "host.id",
+    "host.ip",
+    "host.mac",
+    "host.name",
+    "host.type",
+  ],
+  "history": {
+    "timestampField": "@timestamp",
+    "interval": "1m",
+  },
+  "version": "1.0.0",
+}'
+
+
+
+// CREATE
+ curl -H 'Content-Type: application/json' \
+      -d '{"id":"ea_user_entity_store","name":"EA User Store","type":"node","indexPatterns":["log*"],"filter":"@timestamp >= now-5m","identityFields":["user.name"],"displayNameTemplate":"{{user.name}}","metadata":["user.domain","user.email","user.full_name","user.hash","user.id","user.name","user.roles"],"history":{"timestampField":"@timestamp","interval":"1m"},"version":"1.0.0"}' \
+      -X POST \
+      -H 'kbn-xsrf: true' \
+        -H 'elastic-api-version: 1' \
+      http:///elastic:changeme@localhost:5601/internal/entities/definition
+
+
+// DELETE
+curl -H 'Content-Type: application/json' \
+      -X DELETE \
+      -H 'kbn-xsrf: true' \
+        -H 'elastic-api-version: 1' \
+      http:///elastic:changeme@localhost:5601/internal/entities/definition/todo_host_entity-store
+
+// GET
+curl -H 'Content-Type: application/json' \
+      -X GET \
+      -H 'kbn-xsrf: true' \
+        -H 'elastic-api-version: 1' \
+      http:///elastic:changeme@localhost:5601/internal/entities/definition
+
+
+
+*/
+
+// Blockers
+// We can't access the entity manager plugin from the server side - @kevin Lacambane said he'd look into it https://elastic.slack.com/archives/C043Q05P12B/p1724839105633859
+// We cannot access the latest values only - https://elastic.slack.com/archives/C043Q05P12B/p1724754728872679
```

Comment:
These comments are outdated. Can we delete them and track the progress on issues?

Reply:
ahahha yeah 👍🏽